otx/algorithms/common/tasks/training_base.py

```diff
@@ -347,7 +347,7 @@ def _initialize_post_hook(self, options=None):
         pass
 
     @abc.abstractmethod
-    def _init_recipe(self):
+    def _init_recipe(self, dataset: Optional[DatasetEntity] = None):
```

Comment:
@harimkang I added this parameter to a function to consolidate tiling-related parameters in one place instead of scattering them throughout the codebase. 

However, this change also affects other classes that inherit from this function, as the new parameter is now part of the function's signature and must be supplied when calling it. This requires updating the implementations of those classes to include the new parameter and its functionality.

Reply:
So this is needed for tiling, so we need to add it to the BaseTask as well, right? I don't think there should be any major issues. If we add None as the default for the other tasks as well, shouldn't there be an issue?
@jaegukhyun What do you think?

Reply:
I think there is no harm for overall structure, but I think this could increase complexity of BaseTask, which is complex enough. Let's see... I'll consider how we can refactor for our new task. At the current status, there is no big objection